tests/test_execution.py

```diff
@@ -55,21 +52,19 @@ def test_tuples_execution_on_same_node(global_execution_env):
 
 
 @pytest.mark.slow
-def test_federated_learning_workflow(global_execution_env):
+def test_federated_learning_workflow(factory, clients, client, data_envs):
     """Test federated learning workflow on each node."""
-    factory, initial_assets, network = global_execution_env
-    client = network.clients[0]
 
     # create test environment
     spec = factory.create_algo()
     algo = client.add_algo(spec)
 
     # get first dataset of each client
     # because there is only one dataset created by client, we can get all the datasets
-    datasets = initial_assets.datasets
+    datasets = data_envs.datasets
 
     # check there is one dataset per node in the network
-    assert set([d.owner for d in datasets]) == set([c.node_id for c in network.clients])
+    assert set([d.owner for d in datasets]) == set([c.node_id for c in clients])
```

Comment:
We could even remove this line, the goal of the test is not to check the fixture.

Reply:
You're right, I'll do that too